tabular/src/autogluon/tabular/models/text_prediction/text_prediction_v1_model.py

```diff
@@ -0,0 +1,258 @@
+        return column_properties
+
+    def _get_default_auxiliary_params(self) -> dict:
+        default_auxiliary_params = super()._get_default_auxiliary_params()
+        extra_auxiliary_params = dict(
+            get_features_kwargs=dict(
+                valid_raw_types=[R_INT, R_FLOAT, R_OBJECT],
+                invalid_special_types=[S_TEXT_NGRAM, S_TEXT_AS_CATEGORY, S_TEXT_SPECIAL],
+            ),
+        )
+        default_auxiliary_params.update(extra_auxiliary_params)
+        print('In TextPrediction, default_auxiliary_params=', default_auxiliary_params)
+        return default_auxiliary_params
+
+    def _set_default_params(self):
+        try:
+            from autogluon.text.text_prediction.dataset import TabularDataset
+            from autogluon.text.text_prediction.text_prediction import ag_text_prediction_params
+        except ImportError:
+            raise ImportError(AG_TEXT_IMPORT_ERROR)
+        super()._set_default_params()
+        self.params = ag_text_prediction_params.create('default')
+
+    def _fit(self, X_train: pd.DataFrame, y_train: pd.Series,
+             X_val: Optional[pd.DataFrame] = None,
+             y_val: Optional[pd.Series] = None,
+             time_limit: Optional[int] = None, **kwargs):
+        """The internal fit function
+
+        Parameters
+        ----------
+        X_train
+            Features of the training dataset
+        y_train
+            Labels of the training dataset
+        X_val
+            Features of the validation dataset
+        y_val
+            Labels of the validation dataset
+        time_limit
+            The time limits for the fit function
+        kwargs
+            Other keyword arguments
+
+        """
+        try:
+            from autogluon.text.text_prediction.dataset import TabularDataset
+            from autogluon.text.text_prediction.text_prediction import get_recommended_resource
+        except ImportError:
+            raise ImportError(AG_TEXT_IMPORT_ERROR)
+
+        # Get arguments from kwargs
+        verbosity = kwargs.get('verbosity', 2)
+        num_cpus = kwargs.get('num_cpus', None)
+        num_gpus = kwargs.get('num_gpus', None)
+
+        # Infer resource
+        resource = get_recommended_resource(nthreads_per_trial=num_cpus,
+                                            ngpus_per_trial=num_gpus)
+        X_train = self.preprocess(X_train)
+        X_val = self.preprocess(X_val)
+        column_properties = self._build_model(X_train=X_train,
+                                              y_train=y_train,
+                                              X_val=X_val,
+                                              y_val=y_val,
+                                              hyperparameters=self.params)
+        # Insert the label column
+        X_train[self._label_column_name] = y_train
+        X_val[self._label_column_name] = y_val
+        scheduler_options = self.params['hpo_params']['scheduler_options']
```

Comment:
consider doing params = copy.deepcopy(self.params) to avoid updating objects such as scheduler_options inplace. We don't generally want to update the input self.params values in the fit call.

Reply:
This comment still applies.